src/observerClass.ts

```diff
@@ -20,6 +21,17 @@ export function makeClassComponentObserver(
     componentClass: React.ComponentClass<any, any>
 ): React.ComponentClass<any, any> {
     const target = componentClass.prototype
+
+    if (componentClass[mobxObserverProperty]) {
+        const displayName = getDisplayName(this)
```

Comment:
@ynejati Looks like the issue #887 is related to this change. What is the idea behind using `this` in here?

Reply:
This can't be correct, as `this` isn't bound at this (lol) point, `I think `componentClass.displayName || componentClass.name` will do the trick here

Reply:
In Javascript `"strict"` mode this will die hard iirc

Reply:
Well, that's silly.  Surprised that it's taken this long to catch it.  How embarrassing.  I guess the test failed to catch it as well.  I will submit a PR  later today if someone doesn't get to it before me.

Reply:
`target` seems to work too

Reply:
Test output before (using `this`):

![image](https://user-images.githubusercontent.com/12650339/88544799-d4e1c580-cfce-11ea-8763-faa4519defb2.png)

Test output after (using `target`):

![image](https://user-images.githubusercontent.com/12650339/88544904-ff338300-cfce-11ea-865a-40cd3d892a32.png)